src/components/Header/Header.jsx

```diff
@@ -0,0 +1,133 @@
+import { useState, useEffect, memo } from "react";
+import { FaGithub, FaBars, FaTimes } from "react-icons/fa";
+import { Link } from "react-router-dom";
+import { useMediaQuery } from "react-responsive";
+
+const Header = () => {
+  const [scrollProgress, setScrollProgress] = useState(0);
+  const [isMenuOpen, setIsMenuOpen] = useState(false); // To control the mobile slider
+
+  // Detect if the device is mobile using a media query hook
+  const isMobile = useMediaQuery({ query: "(max-width: 768px)" });
```

Comment:
Just using `sm:` or `lg:` from tailwind should work, I don't think we need to use `react-responsive`

Reply:
did u may check